### PR TITLE
Branch table enroldate

### PR DESCRIPTION
**Pull request by LINCHENYU2030S**
update table feature for enrol date

### CODE REVIEW

File: src/main/java/seedu/address/logic/OrganizeData.java [hunk 1]
```diff
@@ -130,4 +132,40 @@ public static Map<String, Integer> bySubject(Model model) {
...
+                YearMonth yearMonth = subject.getYearMonth();
+                int monthValue = yearMonth.getMonthValue();
+                int yearValue = yearMonth.getYear();
+                if (yearValue <= earliestYearValue && monthValue <= earliestMonthValue) {
+                    earliestMonthValue = monthValue;
+                    earliestYearValue = yearValue;
+                }
+            }
+            if (earliestYearValue == year) {
+                counts[earliestMonthValue]++;
```
Comment (ChuanXinNg) [inline src/main/java/seedu/address/logic/OrganizeData.java:161]: Good implementation on finding the earliest enrol date.

File: src/main/java/seedu/address/logic/commands/tableresults/EnrolDateTableCommandResult.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        this.marCount = columnValueMapping.get("Mar");
+        this.aprCount = columnValueMapping.get("Apr");
+        this.mayCount = columnValueMapping.get("May");
+        this.junCount = columnValueMapping.get("Jun");
+        this.julCount = columnValueMapping.get("Jul");
+        this.augCount = columnValueMapping.get("Aug");
+        this.sepCount = columnValueMapping.get("Sep");
+        this.octCount = columnValueMapping.get("Oct");
+        this.novCount = columnValueMapping.get("Nov");
+        this.decCount = columnValueMapping.get("Dec");
```
Comment (ChuanXinNg) [inline src/main/java/seedu/address/logic/commands/tableresults/EnrolDateTableCommandResult.java:45]: The months can be changed into enumerations, but due to some dependencies, can keep them for now and improve them later after the deadline.

File: src/main/java/seedu/address/logic/parser/TableCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+     * and returns a TableCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public TableCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        args = args.trim();
+        if (args.matches("l/|s/|g/") && args.length() == 2) {
+            return new TableCommand(args);
+        } else if (args.matches("^d/\\d{4}$") && args.length() == 6) {
```
Comment (ChuanXinNg) [inline src/main/java/seedu/address/logic/parser/TableCommandParser.java:26]: Interesting use of regex. Good job!

File: src/main/java/seedu/address/model/tag/Subject.java [hunk 4]
```diff
@@ -49,8 +52,8 @@ private enum Subjects {
             + HIST + ", "
             + "and " + SOC + ".";
 
-    public final String subjectName;
-    public final EnrolDate enrolDate;
+    private final String subjectName;
+    private final EnrolDate enrolDate;
```
Comment (ChuanXinNg) [inline src/main/java/seedu/address/model/tag/Subject.java:56]: Good observation on violation in information hiding.

**Review by ChuanXinNg — APPROVED** [review]
LGTM. Codecov couldn't pass as this update is more on UI. Can merge safely.

**Comment by LINCHENYU2030S** [comment]
The codecov will fail due to missing tests for Ui-related class. Besides that, tests for other classes should be ok. Point out if you observe any issue.

**Pull request merged**